fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java

```diff
@@ -2790,8 +2785,9 @@ private Bundle addLinks(FHIRPagingContext context, Bundle responseBundle, String
 
         // If for search with _summary=count or pageSize == 0, then don't add previous and next links.
         if (!SummaryValueSet.COUNT.equals(summaryParameter) && context.getPageSize() > 0) {
-            int nextPageNumber = context.getPageNumber() + 1;
-            if (nextPageNumber <= context.getLastPageNumber()) {
+            int nextPageNumber = Math.max(context.getPageNumber() + 1, 1);
+            if (nextPageNumber <= context.getLastPageNumber()
```

Comment:
Comment would be useful to describe why we're doing this.

Reply:
Comment added.